configure.ac

```diff
@@ -2391,8 +2387,19 @@ AS_IF([test "x$enable_zph_qos" = "xyes"],[
     [Enable support for QOS netfilter mark preservation])
 ])
 
-AC_CHECK_LIB(regex, regexec, [REGEXLIB="-lregex"],[REGEXLIB=''])
-AC_SUBST(REGEXLIB)
+SQUID_AUTO_LIB(regex,[GNU Regex],[LIBREGEX])
+SQUID_CHECK_LIB_WORKS(regex,[
+  PKG_CHECK_MODULES([LIBREGEX],[libregex],[:],[:])
```

Comment:
This PKG_CHECK_MODULES() addition means that, in environments with the corresponding .pc file, Squid no longer defaults to `-lregex`; Squid now defaults to whatever that .pc file dictates. Please mention this important change/improvement in release notes and commit message. It is more important than the added `--with-regex` support itself in many cases!..